frontend/src/layout/header/UserBox.tsx

```diff
@@ -107,149 +171,189 @@ const WithFloatingMenu: React.FC<{ children: ReactElement }> = ({ children }) =>
-                    display: "flex",
-                    flexDirection: "column",
-                    justifyContent: "center",
-                    lineHeight: 1.3,
-                    paddingRight: 16,
-                    opacity: 0.75,
-                    [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
-                        display: "none",
-                    },
-                }}>
-                    <div css={{ fontSize: 12, color: "var(--grey40)" }}>
-                        {t("user.logged-in-as")}
-                    </div>
-                    <div css={{
-                        flex: "0 1 auto",
-                        textOverflow: "ellipsis",
-                        whiteSpace: "nowrap",
-                        overflow: "hidden",
-                    }}>{user.displayName}</div>
-                </div>
-
-                {/* Show icon */}
-                <ActionIcon title={t("user.settings")}>
-                    <FiUserCheck css={{ "& > polyline": { stroke: "var(--happy-color-dark)" } }}/>
-                </ActionIcon>
-            </div>
-        </WithFloatingMenu>
-    );
+type FloatingMenuProps = {
+    close: () => void;
+    type: MenuType;
 };
 
-
 /**
  * A menu with some user-related settings/actions that floats on top of the page
  * and closes itself on click outside of it.
  */
-const FloatingMenu: React.FC<{ close: () => void }> = ({ close }) => {
+const FloatingMenu: React.FC<FloatingMenuProps> = ({ close, type }) => {
     const { t } = useTranslation();
-
-    type State = "main" | "language";
-    const [state, setState] = useState<State>("main");
-
     const user = useUser();
 
-    const items = match(state, {
+    const handleBlur = (event: React.FocusEvent<HTMLDivElement, Element>) => {
+        if (!event.currentTarget.contains(event.relatedTarget as HTMLDivElement)) {
+            close();
+        }
+    };
+
+    const items = match(type, {
         main: () => <>
-            {/* Login button if the user is NOT logged in */}
-            {user === "none" && (
+            <ReturnButton onClick={() => close()}>{t("User features")}</ReturnButton>
+            {isRealUser(user) && <>
                 <MenuItem
-                    icon={<FiLogIn />}
-                    borderBottom
-                    linkTo={CONFIG.auth.loginLink ?? LOGIN_PATH}
-                    htmlLink={!!CONFIG.auth.loginLink}
                     css={{
-                        color: "var(--nav-color)",
                         [`@media not all and (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
-                            display: "none",
+                            borderRadius: "8px 8px 0 0",
                         },
                     }}
-                >{t("user.login")}</MenuItem>
-            )}
-
-            {isRealUser(user) && <>
+                    icon={<FiFolder />}
+                    borderBottom
+                    linkTo="/~manage"
+                    onClick={() => close()}
```

Comment:
```suggestion
                    onClick={close}
```

And yeah, it has been like this before your change already :P